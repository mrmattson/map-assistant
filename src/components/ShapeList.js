import React from 'react'
import PropTypes from 'prop-types'
import is from 'is_js'

import ShapeListItem from './ShapeListItem'

function ShapeList ({ shapes }) {
  return (
    <ul className="ShapeList list-group">
      {is.not.empty(shapes)
        ? shapes.map(shape => (<ShapeListItem key={shape.properties.id} shape={shape} />))
        : null}
    </ul>
  )
}

// https://facebook.github.io/react/docs/typechecking-with-proptypes.html
ShapeList.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  shapes: PropTypes.array.isRequired
}

export default ShapeList
