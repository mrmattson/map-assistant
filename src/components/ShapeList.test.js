/* eslint-env jest */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import ShapeList from './ShapeList'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <ShapeList
      shapes={[]}
    />, div)
})

it('clean snapshot', () => {
  const component = renderer.create(
    <ShapeList
      shapes={[]}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
