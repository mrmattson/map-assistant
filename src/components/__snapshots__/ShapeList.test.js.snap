exports[`test clean snapshot 1`] = `
<ul
  className="ShapeList list-group" />
`;
